app/Http/Controllers/OfferController.php

```diff
@@ -0,0 +1,101 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Offer;
+use App\Status;
+use Illuminate\Http\Request;
+use Illuminate\Http\Response;
+use Illuminate\Support\Facades\Validator;
+
+class OfferController extends Controller
+{
+    /**
+     * Display a listing of the resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function index()
+    {
+        //
+    }
+
+    /**
+     * Store a newly created resource in storage.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @return \Illuminate\Http\Response
+     */
+    public function store(Request $request)
+    {
+        $validator = Validator::make($request->all(), [
+            "user_id" => "required",
+            "category_id" => "required",
+            "start_at" => "required|date",
+            "end_at" => "required|date|after:start_at",
+            "price_per_hour" => "required|numeric|min:1|max:5000",
+            "address" => "required|max:500",
+            "preferred_qualifications" => "max:500"
+        ]);
+
+        if ($validator->fails()) {
+            return response()->json(["success" => false, "errors" => $validator->errors()], 400);
+        }
+
+        $offer = [
```

Comment:
Laravel has default behavior for binding the attributes from the form data array to the database attributes throw $fillable variable in the model class, so we don't have to bind it manually.

Reply:
I am using mass assignment and to avoid the vulnerability of saving all the request data which may compromise the database, I choose to assign each column in the array separately.
You can find out more about this point of view here: https://laravel.com/docs/7.x/eloquent#mass-assignment